lib/http/request.rb

```diff
@@ -106,7 +106,7 @@ def redirect(uri, verb = @verb)
 
     # Stream the request to a socket
     def stream(socket)
-      include_proxy_headers if using_proxy? && !@uri.https?
```

Comment:
It should be possible according to RFC 2817 to add a Proxy-Authorization header to a CONNECT request, I believe. https://tools.ietf.org/html/rfc2817#section-5.2